### PR TITLE
feat: allow cors on proxy

### PR DESCRIPTION
**Pull request by Benehiko**
Add option to add CORS options to the proxy. This will allow the proxy to add the proper cors headers to requests that require them.

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added necessary documentation within the code base (if
      appropriate).

### CODE REVIEW

**Comment by Benehiko** [comment]
Tests added :)

File: proxy/proxy.go [hunk 1]
```diff
@@ -54,10 +61,13 @@ const (
...
 	return func(r *http.Request) {
-		c, err := o.hostMapper(r.Context(), r)
-		if err != nil {
-			o.onReqError(r, err)
-			return
+		var c *HostConfig
+		var err error
+
+		if oh := r.Context().Value(hostConfigKey); oh == nil {
+			panic("could not get value from context")
```
Comment (aeneasr) [inline proxy/proxy.go:68]: Should we really panic here, or rather call `o.onReqError(r, err)`?

File: proxy/proxy.go [hunk 2]
```diff
@@ -54,10 +61,13 @@ const (
...
+		} else {
+			c = oh.(*HostConfig)
```
Comment (aeneasr) [inline proxy/proxy.go:70]: This could panic!

**Review by aeneasr — CHANGES_REQUESTED** [review]
Just two smal details :)

**Review by aeneasr — APPROVED** [review]

**Review by zepatrik — APPROVED** [review]

**Pull request merged**